### ISSUE TITLE
I can't use it properly on gitlab

### ISSUE THREAD

**Issue by Hargeek** [oauth2-proxy/oauth2-proxy#994]
I want to implement Ingress Nginx + OAuth2-Proxy + GitLab

My environment:
- k8s v1.15.0
- Ingress nginx controller 0.25.0
- image quay.io/oauth2-proxy/oauth2-proxy:latest
- gitlab v10.8.4

My configuration:
```
- --provider=gitlab
- --upstream=file:///dev/null
- --email-domain=*
- --http-address=0.0.0.0:4180
- --cookie-secure=true
- --redirect-url=https://vscode.qa.***.cn/oauth2/callback
- --skip-provider-button=false
- --set-xauthrequest=true
- --skip-auth-preflight=false
- --skip-oidc-discovery
- --oidc-issuer-url=https://git.***.cn
- --client-id=[REDACTED]
- --client-secret=[REDACTED]
- --cookie-secret=[REDACTED]
- --scope=openid email profile groups
- --validate-url=https://git.***.cn/api/v4/user
- --pass-user-headers=true
- --login-url=https://git.***.cn/oauth/authorize
- --redeem-url=https://git.***.cn/oauth/token
- --oidc-jwks-url=https://git.***.cn/oauth/discovery/keys
```

But it's always prompted in GitLab
```
An error occurred
The request scope is invalid, unknown, or incorrectly formatted
```

I really want to solve this problem，It has taken me two days！

**Comment by NickMeves** [comment]
I think people have run into odd issues with the `scope` field in command line options (instead of config files or envvars) because of not escaping the spaces correctly.

I thought the gitlab provider configured its OIDC scope dynamically in the provider. You should try removing that option.

Otherwise I recommend trying a config file or passing it via an environment variable.

**Comment by Hargeek** [comment]
I deleted the configuration with the `scope` field

However, after testing, the problem was not solved, and the same error was reported

**Comment by NickMeves** [comment]
Do you have full logs from the OAuth2 Proxy server (those would give more context with line numbers)? And more details on your request history so I can get a better picture of where you fail in the various OAuth redirect requests in the initial flow?

**Comment by Hargeek** [comment]
Okaythis is the prompt after successful login to the GitLab interface
```

```

**Comment by Hargeek** [comment]
>This is the prompt after a successful login to the GitLab interface

```
An error occurred
The request scope is invalid, unknown, or incorrectly formatted
```

>This is the log for Ingress Nginx

```
172.16.0.160 - [172.16.0.160] - - [17/Jan/2021:17:03:25 +0000] "GET / HTTP/1.1" 308 168 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.16; rv:84.0) Gecko/20100101 Firefox/84.0" 389 0.000 [kube-system-nginx-80] [] - - - - 0973e331bc19a11d7c89928156e31bb5
172.16.0.160 - [172.16.0.160] - - [17/Jan/2021:17:03:25 +0000] "GET /oauth2/auth HTTP/1.1" 401 21 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.16; rv:84.0) Gecko/20100101 Firefox/84.0" 592 0.002 [kube-system-oauth2-proxy-4180] [] 172.20.6.110:4180 21 0.003 401 e0ee9e49abd1e6ea3c8ea029e9ee357b
172.16.0.160 - [172.16.0.160] - - [17/Jan/2021:17:03:25 +0000] "GET / HTTP/2.0" 401 0 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.16; rv:84.0) Gecko/20100101 Firefox/84.0" 0 0.008 [kube-system-nginx-80] [] 172.16.1.110:443 0 0.008 401 3484d9eac28cfd514c977cffd95404f3
172.16.0.160 - [172.16.0.160] - - [17/Jan/2021:17:03:25 +0000] "GET / HTTP/2.0" 302 142 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.16; rv:84.0) Gecko/20100101 Firefox/84.0" 240 0.008 [kube-system-nginx-80] [] - - - - 3484d9eac28cfd514c977cffd95404f3
172.16.0.160 - [172.16.0.160] - - [17/Jan/2021:17:03:25 +0000] "GET /oauth2/start?rd=%2F HTTP/2.0" 302 333 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.16; rv:84.0) Gecko/20100101 Firefox/84.0" 26 0.002 [kube-system-oauth2-proxy-4180] [] 172.20.6.110:4180 333 0.002 302 124cc6d0ec88f9b7eb4eb6424af4d0b1
```

>This is the log of OAuth2 Proxy Server

The test.abc.com is my own testing service

```
[2021/01/17 16:57:21] [logger.go:490] mapping path "/dev/null" => file system "/dev/null"
[2021/01/17 16:57:21] [logger.go:490] OAuthProxy configured for GitLab Client ID: a56bae087601da6d34ebbe459c5c551f6913963bf3db537c8b37ce1752fb2db5
[2021/01/17 16:57:21] [logger.go:490] Cookie settings: name:_oauth2_proxy secure(https):false httponly:true expiry:168h0m0s domains: path:/ samesite: refresh:disabled
[2021/01/17 16:57:21] [logger.go:490] HTTP: listening on 0.0.0.0:4180
[2021/01/17 17:03:25] [logger.go:508] Error loading cookied session: cookie "_oauth2_proxy" not present, removing session
172.16.1.110:12916 - - [2021/01/17 17:03:25] test.abc.com GET - "/oauth2/auth" HTTP/1.1 "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.16; rv:84.0) Gecko/20100101 Firefox/84.0" 401 21 0.000
172.16.1.110:20599 - - [2021/01/17 17:03:25] test.abc.com GET - "/oauth2/start?rd=%2F" HTTP/1.1 "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.16; rv:84.0) Gecko/20100101 Firefox/84.0" 302 333 0.000
```

**Comment by NickMeves** [comment]
Yep, something seems wrong with the `scope` parameter you are sending to Gitlab during the OAuth flow then.

@papey Have you encountered anything like this? I know you recently made some modifications to the gitlab scope for how projects authorization works.

**Comment by papey** [comment]
> @papey Have you encountered anything like this? I know you recently made some modifications to the gitlab scope for how projects authorization works.

No but looking at the Gitlab version, 10.X.X if from 2017, some API changes may have happen since this release.

**Comment by Hargeek** [comment]
But I install the latest Jenkins and login through Gitlab is no problem
I don't think this issue has much to do with the Gitlab version

**Comment by papey** [comment]
Let me rephrase,

The current implementation was tested against a non obsolete Gitlab version (12.X). Some API endpoints may differ from version 10.X and 12.X, causing this error.

**Comment by Hargeek** [comment]
Thanks a lot. I have time to upgrade Gitlab and try again

**Comment by Hargeek** [comment]
Thank you very much. I successfully tested with the latest Gitlab 13.7.4. Thank you!

**Comment by papey** [comment]
@NickMeves do you think it's worth adding some compatibility notes ?

**Comment by NickMeves** [comment]
> Nice !
> 
> @NickMeves do you think it's worth adding some compatibility notes ?

@papey Might be a good idea to add to the Gitlab provider documentation area if there's version constraints with some of the new capabilities. Do you mind opening a PR with the specifics added to the documentation?
